src/components/LocaleContextProvider.tsx

```diff
@@ -0,0 +1,149 @@
+};
+
+const LocaleContext = createContext<LocaleContextProps>({
+    translate: () => '',
+    numberFormat: () => '',
+    datetimeToRelative: () => '',
+    datetimeToCalendarTime: () => '',
+    updateLocale: () => '',
+    formatPhoneNumber: () => '',
+    toLocaleDigit: () => '',
+    fromLocaleDigit: () => '',
+    preferredLocale: CONST.LOCALES.DEFAULT,
+});
+
+function LocaleContextProvider({preferredLocale = CONST.LOCALES.DEFAULT, currentUserPersonalDetails = {}, children}: LocaleContextProviderProps) {
+    const selectedTimezone = useMemo(() => currentUserPersonalDetails?.timezone?.selected, [currentUserPersonalDetails]);
+
+    const translate = useMemo<Translate>(() => (phrase, variables) => Localize.translate(preferredLocale, phrase, variables), [preferredLocale]);
+
+    const numberFormat = useMemo<NumberFormat>(() => (number, options) => NumberFormatUtils.format(preferredLocale, number, options), [preferredLocale]);
+
+    const datetimeToRelative = useMemo<DatetimeToRelative>(() => (datetime) => DateUtils.datetimeToRelative(preferredLocale, datetime), [preferredLocale]);
+
+    const datetimeToCalendarTime = useMemo<DatetimeToCalendarTime>(
+        () =>
+            (datetime, includeTimezone, isLowercase = false) =>
+                DateUtils.datetimeToCalendarTime(preferredLocale, datetime, includeTimezone, selectedTimezone, isLowercase),
+        [preferredLocale, selectedTimezone],
+    );
+
+    const updateLocale = useMemo<UpdateLocale>(() => () => DateUtils.setLocale(preferredLocale), [preferredLocale]);
+
+    const formatPhoneNumber = useMemo<FormatPhoneNumber>(() => (phoneNumber) => LocalePhoneNumber.formatPhoneNumber(phoneNumber), []);
+
+    const toLocaleDigit = useMemo<ToLocaleDigit>(() => (digit) => LocaleDigitUtils.toLocaleDigit(preferredLocale, digit), [preferredLocale]);
+
+    const fromLocaleDigit = useMemo<FromLocaleDigit>(() => (localeDigit) => LocaleDigitUtils.fromLocaleDigit(preferredLocale, localeDigit), [preferredLocale]);
+
+    const contextValue = useMemo(
+        () => ({
+            translate,
+            numberFormat,
+            datetimeToRelative,
+            datetimeToCalendarTime,
+            updateLocale,
+            formatPhoneNumber,
+            toLocaleDigit,
+            fromLocaleDigit,
+            preferredLocale,
+        }),
+        [translate, numberFormat, datetimeToRelative, datetimeToCalendarTime, updateLocale, formatPhoneNumber, toLocaleDigit, fromLocaleDigit, preferredLocale],
+    );
+
+    return <LocaleContext.Provider value={contextValue}>{children}</LocaleContext.Provider>;
+}
+const Provider = compose(
+    withCurrentUserPersonalDetails,
+    withOnyx<LocaleContextProviderProps, LocaleContextProviderOnyxProps>({
+        preferredLocale: {
+            key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+            selector: (preferredLocale) => preferredLocale ?? CONST.LOCALES.DEFAULT,
+        },
+    }),
+)(LocaleContextProvider);
+
```

Comment:
```suggestion

Provider.displayName = 'withOnyx(LocaleContextProvider)';

```

I think we should keep this.